### PR TITLE
[saucer] update to 2.1.0

### PR DESCRIPTION
**Pull request by jimwang118**
Update to 2.1.0

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

### CODE REVIEW

**Comment by Neumann-A** [comment]
Can somebody please train the support crew in the maintainer guidelines? 
This directly violates https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-features-to-implement-alternatives. 
Furthermore this would require Qt5 as a default dependency on linux.....

File: ports/saucer/portfile.cmake [hunk 1]
```diff
@@ -11,10 +11,15 @@ vcpkg_from_github(
         fix-source-generation.patch
 )
 
+set(BACKEND_OPTION)
 if(VCPKG_TARGET_IS_WINDOWS)
-    set(BACKEND_OPTION "-Dsaucer_backend=WebView2")
+    if("webview2" IN_LIST FEATURES)
+        set(BACKEND_OPTION "-Dsaucer_backend=WebView2")
+    endif()
```
Comment (Neumann-A) [inline ports/saucer/portfile.cmake:18]:
```suggestion
    if("webview2" IN_LIST FEATURES)
        set(BACKEND_OPTION "-Dsaucer_backend=WebView2")
    endif()
```

This does nothing. The default backend on windows is WebView2 https://github.com/saucer/saucer/blob/3073fc169f2d2736de115f20c90451068b91f76b/CMakeLists.txt#L58

Comment (jimwang118) [reply ports/saucer/portfile.cmake]: Yes, webview2 is installed by default on windows platform, I will remove it.

File: ports/saucer/portfile.cmake [hunk 2]
```diff
@@ -11,10 +11,11 @@ vcpkg_from_github(
         fix-source-generation.patch
 )
 
-if(VCPKG_TARGET_IS_WINDOWS)
-    set(BACKEND_OPTION "-Dsaucer_backend=WebView2")
-else()
-    set(BACKEND_OPTION "-Dsaucer_backend=Qt6")
+set(BACKEND_OPTION)
```
Comment (Curve) [inline ports/saucer/portfile.cmake:14]: This line is redundant

File: ports/saucer/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "saucer",
   "version": "1.0.1",
```
Comment (Curve) [inline ports/saucer/vcpkg.json:3]: Saucer also has version v2.0.0 now, maybe this could be updated?

Comment (jimwang118) [reply ports/saucer/vcpkg.json:3]: OK, I'll update.

Comment (Curve) [reply ports/saucer/vcpkg.json:3]: > OK, I'll update.

Amazing, thanks!

**Comment by Curve** [comment]
> Can somebody please train the support crew in the maintainer guidelines? 
> This directly violates https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-features-to-implement-alternatives. 
> Furthermore this would require Qt5 as a default dependency on linux.....

If it is not possible at all to leave the backend selection to the user without any hassle I'd be fine with falling back to Qt6 by default.

Will vcpkg ever offer a good way to do this?

**Comment by Neumann-A** [comment]
> Will vcpkg ever offer a good way to do this?

Probably. See https://github.com/microsoft/vcpkg/pull/22216 but it is stuck since https://github.com/microsoft/vcpkg/pull/31748 is first in the backlog

**Comment by jimwang118** [comment]
@Curve I got the following error when updating this port. It prompts that some types are undefined. Where are these types defined?
```
src\webview.webview2.impl.cpp(121): error C2065: 'ICoreWebView2EnvironmentOptions4': undeclared identifier
src\webview.webview2.impl.cpp(121): error C2923: 'Microsoft::WRL::ComPtr': 'ICoreWebView2EnvironmentOptions4' is not a valid template type argument for parameter 'T'
src\webview.webview2.impl.cpp(121): error C2641: cannot deduce template arguments for 'Microsoft::WRL::ComPtr'
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<U> &&,EnableIf<Microsoft::WRL::Details::IsConvertible<U*,T*>::value,void*>::type *) noexcept': expects 2 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<T> &&) noexcept': expects 1 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(const Microsoft::WRL::ComPtr<U> &,EnableIf<Microsoft::WRL::Details::IsConvertible<U*,T*>::value,void*>::type *) noexcept': expects 2 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(const Microsoft::WRL::ComPtr<T> &) noexcept': expects 1 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(U *) noexcept': expects 1 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(nullptr) noexcept': expects 1 arguments - 0 provided
src\webview.webview2.impl.cpp(121): error C2783: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(void) noexcept': could not deduce template argument for 'T'
src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<T>)': expects 1 arguments - 0 provided
src\webview.webview2.impl.cpp(123): error C2787: 'Microsoft::WRL::ComPtr': no GUID has been associated with this object
src\webview.webview2.impl.cpp(123): error C2672: 'IID_PPV_ARGS_Helper': no matching overloaded function found
src\webview.webview2.impl.cpp(123): error C2660: 'Microsoft::WRL::Details::RuntimeClassImpl<Microsoft::WRL::RuntimeClassFlags<2>,true,false,true,CoreWebView2EnvironmentOptionsBase<allocate_fn_t,LPVOID CoTaskMemAlloc(SIZE_T),deallocate_fn_t,void CoTaskMemFree(LPVOID)>>::QueryInterface': function does not take 1 arguments
src\webview.webview2.impl.cpp(129): error C2065: 'CoreWebView2CustomSchemeRegistration': undeclared identifier
src\webview.webview2.impl.cpp(129): error C2672: 'Microsoft::WRL::Details::Make': no matching overloaded function found
src\webview.webview2.impl.cpp(131): error C3536: 'scheme': cannot be used before it is initialized
src\webview.webview2.impl.cpp(132): error C2065: 'ICoreWebView2CustomSchemeRegistration': undeclared identifier
src\webview.webview2.impl.cpp(132): error C2065: 'registration': undeclared identifier
src\webview.webview2.impl.cpp(134): error C2839: invalid return type 'T *' for overloaded 'operator ->'
src\webview.webview2.impl.cpp(134): error C2039: 'SetCustomSchemeRegistrations': is not a member of 'Microsoft::WRL::ComPtr'
src\webview.webview2.impl.cpp(134): error C2065: 'registration': undeclared identifier
```

**Comment by Curve** [comment]
> @Curve The following error occurred while updating this port. It reminds me that some types are undefined. Can you help me check where these types are defined?
> ```
> src\webview.webview2.impl.cpp(121): error C2065: 'ICoreWebView2EnvironmentOptions4': undeclared identifier
> src\webview.webview2.impl.cpp(121): error C2923: 'Microsoft::WRL::ComPtr': 'ICoreWebView2EnvironmentOptions4' is not a valid template type argument for parameter 'T'
> src\webview.webview2.impl.cpp(121): error C2641: cannot deduce template arguments for 'Microsoft::WRL::ComPtr'
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<U> &&,EnableIf<Microsoft::WRL::Details::IsConvertible<U*,T*>::value,void*>::type *) noexcept': expects 2 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<T> &&) noexcept': expects 1 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(const Microsoft::WRL::ComPtr<U> &,EnableIf<Microsoft::WRL::Details::IsConvertible<U*,T*>::value,void*>::type *) noexcept': expects 2 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(const Microsoft::WRL::ComPtr<T> &) noexcept': expects 1 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(U *) noexcept': expects 1 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(nullptr) noexcept': expects 1 arguments - 0 provided
> src\webview.webview2.impl.cpp(121): error C2783: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(void) noexcept': could not deduce template argument for 'T'
> src\webview.webview2.impl.cpp(121): error C2780: 'Microsoft::WRL::ComPtr<T> Microsoft::WRL::ComPtr(Microsoft::WRL::ComPtr<T>)': expects 1 arguments - 0 provided
> src\webview.webview2.impl.cpp(123): error C2787: 'Microsoft::WRL::ComPtr': no GUID has been associated with this object
> src\webview.webview2.impl.cpp(123): error C2672: 'IID_PPV_ARGS_Helper': no matching overloaded function found
> src\webview.webview2.impl.cpp(123): error C2660: 'Microsoft::WRL::Details::RuntimeClassImpl<Microsoft::WRL::RuntimeClassFlags<2>,true,false,true,CoreWebView2EnvironmentOptionsBase<allocate_fn_t,LPVOID CoTaskMemAlloc(SIZE_T),deallocate_fn_t,void CoTaskMemFree(LPVOID)>>::QueryInterface': function does not take 1 arguments
> src\webview.webview2.impl.cpp(129): error C2065: 'CoreWebView2CustomSchemeRegistration': undeclared identifier
> src\webview.webview2.impl.cpp(129): error C2672: 'Microsoft::WRL::Details::Make': no matching overloaded function found
> src\webview.webview2.impl.cpp(131): error C3536: 'scheme': cannot be used before it is initialized
> src\webview.webview2.impl.cpp(132): error C2065: 'ICoreWebView2CustomSchemeRegistration': undeclared identifier
> src\webview.webview2.impl.cpp(132): error C2065: 'registration': undeclared identifier
> src\webview.webview2.impl.cpp(134): error C2839: invalid return type 'T *' for overloaded 'operator ->'
> src\webview.webview2.impl.cpp(134): error C2039: 'SetCustomSchemeRegistrations': is not a member of 'Microsoft::WRL::ComPtr'
> src\webview.webview2.impl.cpp(134): error C2065: 'registration': undeclared identifier
> ```
> 

These types are from WebView2, maybe the supplied WebView2 version is too old?

**Comment by jimwang118** [comment]
> These types are from WebView2, maybe the supplied WebView2 version is too old?

Thank you, it is indeed that the webview2 version I am using is too old.

File: ports/saucer/vcpkg.json [hunk 4]
```diff
@@ -1,14 +1,15 @@
 {
   "name": "saucer",
-  "version": "1.0.1",
-  "port-version": 3,
+  "version": "2.0.0",
```
Comment (Curve) [inline ports/saucer/vcpkg.json:3]: 2.1.0 is also available, there should be no dependency changes

File: ports/saucer/vcpkg.json [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
   "description": "Next-gen desktop apps with web-frontend in C++",
   "homepage": "https://saucer.github.io/",
   "license": "MIT",
   "supports": "!uwp",
   "dependencies": [
```
Comment (Curve) [inline ports/saucer/vcpkg.json:8]: Wil, which is listed as a dependency further down, is no longer used and can be removed

File: ports/saucer/vcpkg.json (hunk 4 above)
Comment (jimwang118) [reply ports/saucer/vcpkg.json:3]: OK, I'll update it.

File: ports/saucer/vcpkg.json (hunk 5 above)
Comment (jimwang118) [reply ports/saucer/vcpkg.json:8]: I will delete `wil`, but I found that you added a new dependency `flagpp`. This is not in our `vcpkg`. Can you add it to `vcpkg`?

Comment (Curve) [reply ports/saucer/vcpkg.json:8]: > I will delete `wil`, but I found that you added a new dependency `flagpp`. This is not in our `vcpkg`. Can you add it to `vcpkg`?

I can submit a PR later

**Review by MonicaLiu0311 — DISMISSED** [review]

File: versions/s-/saucer.json [hunk 6]
```diff
@@ -1,5 +1,20 @@
...
+    {
+      "git-tree": "1b778b1017e199a57866367877c41f5fea56fd99",
+      "version": "2.0.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "5229c7f36803a05c353c7327e06d88f69d5c0033",
+      "version": "1.0.1",
+      "port-version": 4
+    },
```
Comment (JavierMatosD) [inline versions/s-/saucer.json:17]: There shouldn't be three version entries here.

File: ports/lockpp/vcpkg.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lockpp",
-  "version": "1.0.2",
+  "version": "2.5",
```
Comment (JavierMatosD) [inline ports/lockpp/vcpkg.json:3]: It looks like upstream is on version 2.6. Please update to latest

Comment (jimwang118) [reply ports/lockpp/vcpkg.json]: Has been updated to version 2.6.

**Comment by Curve** [comment]
@jimwang118 thanks a lot for your work!

**Pull request merged**